chore: deprecate `emitESMImage` function (#13589)
withastro/astro: .changeset/good-worms-heal.md

---
'@astrojs/markdoc': patch
'astro': patch
---

> The function
> `emitESMImage` has been deprecated, and it will be removed in a next major release of Astro. Use the function
> `emitImageMetadata` instead.

Reviewer: ```suggestion
Deprecates the asset utility function `emitESMImage()` and adds a new `emitImageMetadata()` to be used instead

The function `emitESMImage()` is now deprecated. It will continue to function, but it is no longer recommended nor supported. This function will be completely removed in a next major release of Astro. 

Please replace it with the new function`emitImageMetadata()` as soon as you are able to do so:
```

```diff
- import {emitESMImage} from "astro/assets/utils";
+ import {emitImageMetadata} from "astro/assets/utils";

> ```

Reviewer: ``````suggestion
```

The new function was created to operate the same way as the old function, and should work interchangeably in your code. However, its signature is not identical and you may need to check that your code is working as intended.
``````
Given that this isn't an *identical* function, it might be good to include something like this. If you can be more specific about *what* people might need to check or *where* they might notice something breaking, that's even better!

---
'astro': patch
---

Deprecates the asset utility function `emitESMImage()` and adds a new `emitImageMetadata()` to be used instead

The function `emitESMImage()` is now deprecated. It will continue to function, but it is no longer recommended nor supported. This function will be completely removed in a next major release of Astro. 

Please replace it with the new function`emitImageMetadata()` as soon as you are able to do so:

```diff

> - import {emitESMImage} from "astro/assets/utils";
> + import {emitImageMetadata} from "astro/assets/utils";

Reviewer 2: ```suggestion
- import { emitESMImage } from "astro/assets/utils";
+ import { emitImageMetadata } from "astro/assets/utils";
```

---
'astro': patch
---

Deprecates the asset utility function `emitESMImage()` and adds a new `emitImageMetadata()` to be used instead

The function
`emitESMImage()` is now deprecated. It will continue to function, but it is no longer recommended nor supported. This function will be completely removed in a next major release of Astro.

Please replace it with the new function`emitImageMetadata()` as soon as you are able to do so:

```diff
- import { emitESMImage } from "astro/assets/utils";
+ import { emitImageMetadata } from "astro/assets/utils";
```

> The new function returns the same signature as the previous one, however the new function drops some deprecated arguments that weren't meant to be exposed to users;`_watchMode` and `experimentalSvgEnabled` are removed and you may need to check if your code works as intended. 

Reviewer: ```suggestion
The new function returns the same signature as the previous one. However, the new function removes two deprecated arguments that were not meant to be exposed for public use: `_watchMode` and `experimentalSvgEnabled`.  Since it was possible to access these with the old function, you may need to verify that your code still works as intended with `emitImageMetadata()`.
```
Maybe something like this?

Deprecates the asset utility function `emitESMImage()` and adds a new `emitImageMetadata()` to be used instead

The function
`emitESMImage()` is now deprecated. It will continue to function, but it is no longer recommended nor supported. This function will be completely removed in a next major release of Astro.

Please replace it with the new function`emitImageMetadata()` as soon as you are able to do so:

```diff
- import { emitESMImage } from "astro/assets/utils";
+ import { emitImageMetadata } from "astro/assets/utils";
```

> The new function returns the same signature as the previous one, however the new function drops some deprecated arguments that weren't meant to be exposed to users;`_watchMode` and `experimentalSvgEnabled` are removed and you may need to check if your code works as intended. 

Reviewer 2: ```suggestion
The new function returns the same signature as the previous one, however the new function drops some deprecated arguments that weren't meant to be exposed to users: `_watchMode` and `experimentalSvgEnabled` are removed and you may need to check if your code works as intended. 
```
